cvmfs/ssl.cc

```diff
@@ -0,0 +1,109 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SSL_H_
+#define CVMFS_SSL_H_
+
+#include "ssl.h"
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+
+#include "duplex_curl.h"
+#include "platform.h"
+#include "util/posix.h"
+#include "util/string.h"
+
+#ifdef CVMFS_NAMESPACE_GUARD
+namespace {
+#endif
+
+bool HasCertificates(std::string directory) {
+  if (!DirectoryExists(directory)) return false;
+
+  DIR *dirp = opendir(directory.c_str());
+  if (!dirp) return false;
+
+  platform_dirent64 *dirent;
+  while ((dirent = platform_readdir(dirp))) {
+    platform_stat64 stat;
+    if (0 == platform_stat(dirent->d_name, &stat)) {
```

Comment:
Easier to read if reversed: 

    if (platform_stat(dirent->d_name, &stat) != 0)
      continue;